### ISSUE TITLE
empty LC_* vars from getty@.service leak into user sesssion

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#6407]
### Submission type

 - Bug report

### systemd version the issue has been seen with

v234

### Used distribution

Debian
See downstream bug report
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=868695

### In case of bug report: Expected behaviour you didn't see

After login on tty, locale is properly set to the configured system default.

### In case of bug report: Unexpected behaviour you saw

The empty LC_* and LANGUAGE env vars from getty@.service leak into the user session, breaking the conflgured system locale.

### In case of bug report: Steps to reproduce the problem

Login on tty2, run export and locale
With v233, it's properly set:
```
root@debian:~# locale
LANG=de_DE.UTF-8
LANGUAGE=
LC_CTYPE="de_DE.UTF-8"
LC_NUMERIC="de_DE.UTF-8"
LC_TIME="de_DE.UTF-8"
LC_COLLATE="de_DE.UTF-8"
LC_MONETARY="de_DE.UTF-8"
LC_MESSAGES="de_DE.UTF-8"
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL=
root@debian:~# export
declare -x HOME="/root"
declare -x HUSHLOGIN="FALSE"
declare -x LANG="de_DE.UTF-8"
declare -x LOGNAME="root"
declare -x MAIL="/var/mail/root"
declare -x OLDPWD
declare -x PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x SHELL="/bin/bash"
declare -x SHLVL="2"
declare -x TERM="linux"
declare -x USER="root"
declare -x XDG_RUNTIME_DIR="/run/user/0"
declare -x XDG_SEAT="seat0"
declare -x XDG_SESSION_ID="1"
declare -x XDG_VTNR="1"
root@debian:~# cat /etc/default/locale»
#  File generated by update-locale
LANG="de_DE.UTF-8"
```
With v234, notice the empty LC_ and LANGUAGE vars:
```
root@debian:~# locale
LANG=de_DE.UTF-8
LANGUAGE=
LC_CTYPE=
LC_NUMERIC=
LC_TIME=
LC_COLLATE=
LC_MONETARY=
LC_MESSAGES=
LC_PAPER=
LC_NAME=
LC_ADDRESS=
LC_TELEPHONE=
LC_MEASUREMENT=
LC_IDENTIFICATION=
LC_ALL=
root@debian:~# export
declare -x HOME="/root"
declare -x HUSHLOGIN="FALSE"
declare -x INVOCATION_ID="df1fca1fa610401988750827358aa159"
declare -x JOURNAL_STREAM="9:12556"
declare -x LANG="de_DE.UTF-8"
declare -x LANGUAGE=""
declare -x LC_ADDRESS=""
declare -x LC_COLLATE=""
declare -x LC_CTYPE=""
declare -x LC_IDENTIFICATION=""
declare -x LC_MEASUREMENT=""
declare -x LC_MESSAGES=""
declare -x LC_MONETARY=""
declare -x LC_NAME=""
declare -x LC_NUMERIC=""
declare -x LC_PAPER=""
declare -x LC_TELEPHONE=""
declare -x LC_TIME=""
declare -x LOGNAME="root"
declare -x MAIL="/var/mail/root"
declare -x OLDPWD
declare -x PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x SHELL="/bin/bash"
declare -x SHLVL="2"
declare -x TERM="linux"
declare -x USER="root"
declare -x XDG_RUNTIME_DIR="/run/user/0"
declare -x XDG_SEAT="seat0"
declare -x XDG_SESSION_ID="1"
declare -x XDG_VTNR="1"
root@debian:~# cat /etc/default/locale»
#  File generated by update-locale
LANG="de_DE.UTF-8"
```

**Comment by mbiebl** [comment]
According to the downstream bug report, this breaks tools like gpg-agent

**Comment by mbiebl** [comment]
This was broken by #6023 
/CC @spbnick

**Comment by poettering** [comment]
Hmm, I figure the real bug here is that `Environment=FOO=` doesn't unset the env var as the current unit assumes but actually sets it to the empty string. I figure we need to add UnsetEnvironment= or so, to explicitly remove vars from the environment again.

**Comment by mbiebl** [comment]
Yeah, the comment in getty@.service in that regard is misleading. We should probably fix that while at it
```
# Unset locale for the console getty since the console has problems
# displaying some internationalized messages.
Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONET
```

**Comment by poettering** [comment]
fix waiting in #6790.
